wp_api/src/request.rs

```diff
@@ -213,12 +214,33 @@ impl WpNetworkRequest {
     }
 
     /// Does this request specify HTTP login details of some kind?
-    pub fn has_http_authorization_header(&self) -> bool {
+    pub fn has_http_authentication(&self) -> bool {
         self.header_map
             .inner
             .get(http::header::AUTHORIZATION)
             .is_some()
     }
+
+    pub fn adding_http_authentication(&self, username: &str, password: &str) -> Self {
+        let encoded_credentials =
+            base64::engine::general_purpose::STANDARD.encode(format!("{}:{}", username, password));
+
+        let mut new_header_map = self.header_map.inner.clone();
+
+        new_header_map.insert(
+            http::header::AUTHORIZATION,
+            format!("Basic {}", encoded_credentials).parse().unwrap(), // base64 can only produce ASCII, so this string is guaranteed to be parseable
+        );
+
+        WpNetworkRequest {
+            uuid: self.uuid.clone(),
+            retry_count: self.retry_count,
+            method: self.method.clone(),
+            url: self.url.clone(),
+            header_map: Arc::new(new_header_map.into()),
+            body: self.body.clone(),
+        }
```

Comment:
We should be able to avoid all these clones by making this function take `self`'s ownership.